public/javascripts/tinymce/plugins/imagetools/plugin.min.js

```diff
@@ -0,0 +1 @@
idth,e=b.height,(d||e)&&a.$(b).attr({width:c.w,height:c.h}))}function k(a){return{w:a.naturalWidth,h:a.naturalHeight}}function l(){return a.selection.getNode()}function m(){return"imagetools"+A++}function n(b){var c=b.src;return 0===c.indexOf("data:")||0===c.indexOf("blob:")||new d(c).host===a.documentBaseURI.host}function o(b){return-1!==e.inArray(a.settings.imagetools_cors_hosts,new d(b.src).host)}function p(a){return new c(function(b){var c=new XMLHttpRequest;c.onload=function(){b(this.response)},c.open("GET",a,!0),c.responseType="blob",c.send()})}function q(b){var c=b.src;return o(b)?p(b.src):(n(b)||(b=new Image,c=a.settings.imagetools_proxy,b.src+=(-1===c.indexOf("?")?"?":"&")+"url="+encodeURIComponent(b.src)),g.imageToBlob(b))}function r(){var b;return b=a.editorUpload.blobCache.getByUri(l().src),b?b:q(l()).then(function(b){return g.blobToBase64(b).then(function(c){var d=a.editorUpload.blobCache,e=d.create(m(),b,c);return d.add(e),e})})}function s(b){return g.blobToDataUri(b).then(function(c){var e,f,g,h,i;return i=l(),e=m(),g=a.editorUpload.blobCache,f=d.parseDataUri(c).data,h=g.create(e,b,f),g.add(h),a.undoManager.transact(function(){function b(){a.$(i).off("load",b),a.nodeChanged()}a.$(i).on("load",b),a.$(i).attr({src:h.blobUri()}).removeAttr("data-mce-src")}),h})}function t(b){return function(){return a._scanForImages().then(r).then(b).then(s)}}function u(a){return function(){return t(function(b){var c=i(l());return c&&j(l(),{w:c.h,h:c.w}),f.rotate(b.blob(),a)})()}}function v(a){return function(){return t(function(b){return f.flip(b.blob(),a)})()}}function w(){var a=l(),b=k(a);a&&q(a).then(h.edit).then(function(d){return new c(function(c){g.blobToImage(d).then(function(e){var f=k(e);(b.w!=f.w||b.h!=f.h)&&i(a)&&j(a,f),URL.revokeObjectURL(e.src),c(d)})})}).then(s,function(){})}function x(){a.addButton("rotateleft",{title:"Rotate counterclockwise",onclick:u(-90)}),a.addButton("rotateright",{title:"Rotate clockwise",onclick:u(90)}),a.addButton("flipv",{title:"Flip vertically",onclick:v("v")}),a.addButton("fliph",{title:"Flip horizontally",onclick:v("h")}),a.addButton("editimage",{title:"Edit image",onclick:w}),a.addButton("imageoptions",{title:"Image options",icon:"options",cmd:"mceImage"})}function y(b){var c=a.dom.is(b,"img:not([data-mce-object],[data-mce-placeholder])");return c&&(n(b)||o(b)||a.settings.imagetools_proxy)}function z(){var b=a.settings.imagetools_toolbar;b||(b="rotateleft rotateright | flipv fliph | crop editimage imageoptions"),a.addContextToolbar(y,b)}var A=0;b.fileApi&&(x(),z())})}),f(["tinymce/imagetoolsplugin/Canvas","tinymce/imagetoolsplugin/Mime","tinymce/imagetoolsplugin/ImageSize","tinymce/imagetoolsplugin/Conversions","tinymce/imagetoolsplugin/ImageTools","tinymce/imagetoolsplugin/CropRect","tinymce/imagetoolsplugin/ImagePanel","tinymce/imagetoolsplugin/ColorMatrix","tinymce/imagetoolsplugin/Filters","tinymce/imagetoolsplugin/UndoStack","tinymce/imagetoolsplugin/Dialog","tinymce/imagetoolsplugin/Plugin"])}(this);
```

Comment:
Line is too long.<br>'c' is already defined.<br>Expected '{' and instead saw 'throw'.<br>Missing semicolon.<br>'d' is already defined.<br>'e' is already defined.<br>Expected '{' and instead saw 'return'.<br>'f' is already defined.<br>Expected '{' and instead saw 'e'.<br>Expected an assignment or function call and instead saw an expression.<br>'a' is already defined.<br>'b' is already defined.<br>A constructor name should start with an uppercase letter.<br>Missing '()' invoking a constructor.<br>Too many errors. (100% scanned).